src/main/scala/no/ndla/articleapi/repository/ArticleRepository.scala

```diff
@@ -95,27 +157,37 @@ trait ArticleRepository {
     }
 
     def getExternalIdsFromId(id: Long)(implicit session: DBSession = AutoSession): List[String] = {
-      sql"select external_id from ${Article.table} where id=${id.toInt}"
+      sql"""
+           select external_id
+           from ${Article.table}
+           where article_id=${id.toInt}
+           order by revision desc
+           limit 1
+         """
         .map(externalIdsFromResultSet)
         .single()
         .apply()
         .getOrElse(List.empty)
     }
 
     def getAllIds(implicit session: DBSession = AutoSession): Seq[ArticleIds] = {
-      sql"select id, external_id from ${Article.table}"
+      sql"select article_id, external_id from ${Article.table}"
         .map(
           rs =>
             ArticleIds(
-              rs.long("id"),
+              rs.long("article_id"),
               externalIdsFromResultSet(rs)
           ))
         .list()
         .apply()
     }
 
     def articleCount(implicit session: DBSession = AutoSession): Long = {
-      sql"select count(*) from ${Article.table} where document is not NULL"
+      sql"""
+           select count(distinct article_id)
+           from ${Article.table}
+           where document is not NULL
```

Comment:
Denne inkluderer artikler som for eksempel har siste versjon avpublisert. Veit ikkje om vi burde endra på det?

Reply:
Hmm, burde vel kanskje endres ja siden vi bare indekserer siste revision i search-api.